### PR TITLE
Hotfix/implement and fixed unit tests for application manager in genivi

### PR DESCRIPTION
**Pull request by monstersem**
Fixed unit tests in ApplicationManager.

### CODE REVIEW

**Comment by monstersem** [comment]
Please rewiew: @AGaliuzov , @Kozoriz, @iAndrew5, @Anatoliy-Leshin

File: src/components/application_manager/src/application_state.cc [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  switch (state->state_id()) {
+  case HmiState::StateID::STATE_ID_REGULAR:
+    SetRegularState(state);
+    return;
+  case HmiState::StateID::STATE_ID_POSTPONED:
+    SetPostponedState(state);
+    return;
+//  case HmiState::StateID::STATE_ID_CURRENT:
+//    LOG4CXX_ERROR(logger_, "State of type '" << state << "' can't be added.");
+//    return;
```
Comment (iAndrew5) [inline src/components/application_manager/src/application_state.cc]: 3 lines commented code... Find out if it needed then remove or uncomment.

File: src/components/application_manager/src/application_state.cc [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  return;
+}
+
+void ApplicationState::RemoveState(HmiState::StateID state) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(state);
+  switch (state) {
+  //case HmiState::StateID::STATE_ID_CURRENT:
```
Comment (iAndrew5) [inline src/components/application_manager/src/application_state.cc]: commented code...

File: src/components/application_manager/src/application_state.cc [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+HmiStatePtr ApplicationState::PostponedHmiState() const {
+  sync_primitives::AutoLock auto_lock(hmi_states_lock_);
+  DCHECK_OR_RETURN(!hmi_states_.empty(), HmiStatePtr());
+  HmiStatePtr front_state = hmi_states_.front();
+  return front_state->state_id() == HmiState::StateID::STATE_ID_POSTPONED ?
+         front_state : HmiStatePtr();
+}
+
+} // namespace application_manager
```
Comment (iAndrew5) [inline src/components/application_manager/src/application_state.cc]: 2 spaces between code & comment

**Comment by Kozoriz** [comment]
Reviewed

**Pull request merged**